test/screens/play/game_end_page_test.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:treasure_of_the_high_seas/screens/play/game_end_page.dart';
+import 'package:treasure_of_the_high_seas/screens/play/play_page.dart';
+import 'package:treasure_of_the_high_seas/screens/play/player_hand.dart';
+
+import '../../test_utils.dart';
+
+void main() {
+  testWidgets('should display correct text for win and loss', (WidgetTester tester) async {
+    await tester.pumpWidget(createWidgetForTesting(child: GameEndPage(GameResult.WIN)));
+    expect(find.byType(GameEndPage), findsOneWidget);
+    expect(find.byType(PlayerHand), findsNothing);
+
+    var gameWinTextFinder = find.text(GameResult.WIN.description);
+    var gameLoseTextFinder = find.text(GameResult.LOSE.description);
+    expect(gameWinTextFinder, findsOneWidget);
+    expect(gameLoseTextFinder, findsNothing);
+
+    await tester.pumpWidget(createWidgetForTesting(child: GameEndPage(GameResult.LOSE)));
+    expect(find.byType(GameEndPage), findsOneWidget);
+    expect(find.byType(PlayerHand), findsNothing);
+
+    gameWinTextFinder = find.text(GameResult.WIN.description);
+    gameLoseTextFinder = find.text(GameResult.LOSE.description);
+    expect(gameWinTextFinder, findsNothing);
+    expect(gameLoseTextFinder, findsOneWidget);
+  });
+
+  testWidgets('new game button should start a new game', (WidgetTester tester) async {
+    await tester.pumpWidget(createWidgetForTesting(child: GameEndPage(GameResult.WIN)));
+    expect(find.byType(GameEndPage), findsOneWidget);
+    expect(find.byType(PlayerHand), findsNothing);
+
+    final button1Finder = find.text('New Game');
+    expect(button1Finder, findsOneWidget);
+
+    await tester.tap(button1Finder);
+    await tester.pumpAndSettle();
+    expect(find.byType(PlayerHand), findsOneWidget);
+    expect(find.text('Rules'), findsNothing);
+
+    final playPage = tester.widget<PlayPage>(find.byType(PlayPage));
+    final gameState = playPage.state;
+    expect(gameState.result, null);
+    expect(gameState.playerHand.cards, INITIAL_RESOURCES);
+    expect(gameState.currentCard, isNotNull);
+  });
+
+  testWidgets('exit button should start exit to main menu', (WidgetTester tester) async {
+    await tester.pumpWidget(createWidgetForTesting(child: MainMenuPage()));
+
+    final button1Finder = find.text('Play');
+    expect(button1Finder, findsOneWidget);
+    await tester.tap(button1Finder);
+    await tester.pumpAndSettle();
+
+    final playPage = tester.widget<PlayPage>(find.byType(PlayPage));
+    final gameState = playPage.state;
+
+    gameState.endGame(GameResult.LOSE);
+    await tester.pumpAndSettle();
+    expect(find.byType(GameEndPage), findsOneWidget);
+    expect(find.byType(PlayerHand), findsNothing);
+
+    final button2Finder = find.text('Exit');
+    expect(button2Finder, findsOneWidget);
+
+    await tester.tap(button2Finder);
+    await tester.pumpAndSettle();
+    expect(find.byType(PlayerHand), findsNothing);
+    expect(find.text('Rules'), findsOneWidget);
+  });
```

Comment:
Feels unfortunate that we have to go via the menu to test `pop` - wondering if there might be another way of testing this. Thinking we might be able to stick a `MenuScreen` onto the back stack at the point where we create the App widget - at which point we can just spin up a `GameEndPage` and hit the button and be done in a more targeted way.

At the very least, I guess I'd prefer a `find.byType(MainMenuPage)` at the end (rather than looking for the text "Rules"). Maybe this is just the way we have to test it - doesn't feel the end of the world, it's just touching more components than you'd typically expect for an individual test.

Reply:
So is that a change to the main program or just how we generate the test?

Reply:
Was just thinking we could short-circuit the test somehow to make it shorter, but haven't found anything in a brief google and this seems to be the strategy we've used at work for testing `pop`, so looks good.